src/main/java/com/cloudbees/plugins/credentials/UserCredentialsProvider.java

```diff
@@ -145,8 +145,8 @@
                 UserCredentialsProperty property = user.getProperty(UserCredentialsProperty.class);
                 if (property != null) {
                     // we need to impersonate if the requesting authentication is not the current authentication.
-                    boolean needImpersonation = user.equals(User.current());
-                    SecurityContext old = needImpersonation ? null : ACL.impersonate(user.impersonate());
+                    boolean needImpersonation = !user.equals(User.current());
+                    SecurityContext old = needImpersonation ? ACL.impersonate(user.impersonate()) : null;
```

Comment:
Maybe I am being blinded, but is this just a no-op, equivalent to renaming the variable?